File: algoritmos/potencia_sin_potencia/ejemploV1.py
def exponentiation_v1(num1, num2):
    result = 1

    if (num1 == 0 and num2 == 0) or num2 == 0:
        return result
    elif num1 == 0:
        return 0

    while num2:
        num2 -= 1
        result *= num1

    return result


def run():
    pass


if __name__ == "__main__":
    run()

File: algoritmos/potencia_sin_potencia/ejemploV1.js
const exponentiationV1 = (num1, num2) => {
  let result = 1;

  if ((num1 === 0 && num2 === 0) || num2 === 0) {
    return result;
  } else if (num1 === 0) {
    return 0;
  }

  while (num2) {
    num2--;
    result *= num1;
  }

  return result;
};
